.github/workflows/javascript-tests.yaml

```diff
@@ -9,12 +9,16 @@ on:
     types:
       - opened
       - labeled
+      - synchronize
 
 
 jobs:
   test:
     runs-on: ubuntu-latest
-    if: contains(github.event.pull_request.labels.*.name, 'okay-to-test') || (github.event_name == 'push')
+    if: |
+      (contains(github.event.pull_request.labels.*.name, 'okay-to-test') && github.event.action == 'labeled') ||
+      contains(fromJson('["MEMBER", "COLLABORATOR"]'), github.event.pull_request.author_association) ||
+      github.event_name == 'push'
```

Comment:
Wouldn't this if statement always run this action on push since it's an or statement?

It looks like the current logic is run the workflow in any of these conditions:
1. The event is labeled with okay to test
2. The author is a member or collaborator
3. The event is push

Based on my understanding of this issue, for this to be safe I think we would need the following logic:
1. The PR is labeled with okay-to-test by a trusted collaborator / maintainer
2. The event is push and the commit author (not the pull request author?) is a trusted collaborator / maintainer

Reply:
This job would run on pushes, but this workflow only runs on pushes to the main branch. When a PR author pushes to the source branch on the PR, then `github.event_name` is `pull_request_target` and `github.event.action` is `synchronize`, which means the conditions would not be met for the job to run. So I believe it's safe as is. I'll add some comments though to make it clearer.

Reply:
That makes sense, thanks for the explanation. I haven't used this flow before, so just wanted to make sure I was understanding what the goal was, and how this was working. I also learned that adding tags is a restricted action, which was a piece that I was missing. Thanks for your help!